nnpdf_data/nnpdf_data/commondata/CMS_WCHARM_13TEV/filter_utils.py

```diff
@@ -0,0 +1,304 @@
+import logging
+import os
+
+import numpy as np
+import pandas as pd
+from sys_uncertainties import SYS_DEFINITIONS, SYS_UNC_BY_BIN
+import yaml
+
+from nnpdf_data.filter_utils.utils import prettify_float, symmetrize_errors
+
+yaml.add_representer(float, prettify_float)
+
+SQRTS = 8000
+MW2 = 80.385**2
+CMSLUMI13 = 2.5  # %
+
+# List of systematic uncertainties that shuold
+# be considered uncorrelated
+UNCORR_SYS_UNC = ['UnfoldMCstat', 'UnfoldOtherGen', 'UnfoldReweight']
+ART_LABEL = 'art_corr_unc'
+STAT_LABEL = 'stat_uncorr_unc'
+TABLE = ''
+
+# From Table 1 of the paper
+SYS_UNC_by_bin = [{}]
+
+
+class Extractor:
```

Comment:
Some of this stuff seems pretty universal. Would it be worth defining a base class that is shared between datasets?

Reply:
I've been thinking about making this class more universal. However, I gave up because there are many differences between datasets, even within the same experiment. If we want to use a more universal `extractor`, then we should all agree on standard common specifics amongst datasets. For now, the extractor class is rather specialised to the datasets that I implemented.